lib/wallet/src/Cardano/Wallet/Write/Tx.hs

```diff
@@ -491,9 +617,74 @@ utxoFromTxOutsInLatestEra
 utxoFromTxOutsInLatestEra = withStandardCryptoConstraint RecentEraBabbage $
     Shelley.UTxO . Map.fromList . map (second castTxOut)
 
+--------------------------------------------------------------------------------
+-- Tx
+--------------------------------------------------------------------------------
+
+modifyTxOutputs
+    :: RecentEra era
+    -> (TxOut (ShelleyLedgerEra era) -> TxOut (ShelleyLedgerEra era))
+    -> Core.TxBody (ShelleyLedgerEra era)
+    -> Core.TxBody (ShelleyLedgerEra era)
+modifyTxOutputs era f body = case era of
+    RecentEraBabbage -> body
+        { Babbage.outputs = mapSized f <$> Babbage.outputs body
+        }
+    RecentEraAlonzo -> body
+        { Alonzo.outputs = f <$> Alonzo.outputs body
+        }
+  where
+    mapSized f' = mkSized . f' . sizedValue
+
+txBody
+    :: RecentEra era
+    -> Core.Tx (ShelleyLedgerEra era)
+    -> Core.TxBody (ShelleyLedgerEra era)
+txBody RecentEraBabbage = Alonzo.body -- same type for babbage
+txBody RecentEraAlonzo = Alonzo.body
+
+-- Until we have convenient lenses to use
+outputs
+    :: RecentEra era
+    -> Core.TxBody (ShelleyLedgerEra era)
+    -> [TxOut (ShelleyLedgerEra era)]
+outputs RecentEraBabbage = map sizedValue . toList . Babbage.outputs
+outputs RecentEraAlonzo = toList . Alonzo.outputs
```

Comment:
Since this function can only extract outputs from a `TxBody`:

```suggestion
txBodyOutputs
    :: RecentEra era
    -> Core.TxBody (ShelleyLedgerEra era)
    -> [TxOut (ShelleyLedgerEra era)]
txBodyOutputs RecentEraBabbage = map sizedValue . toList . Babbage.outputs
txBodyOutputs RecentEraAlonzo = toList . Alonzo.outputs
```